venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js

```diff
@@ -0,0 +1,54 @@
+import { firstName, lastName, accountEmail, accountPassword } from '../../../fixtures/accountAccess/index';
+
+import { createAccount, openLoginDialog } from '../../../actions/accountAccess/index';
+
+import { visitPage } from '../../../actions/routes/index';
+
+import { assertCreateAccount, goToMyAccount } from '../../../actions/myAccountMenu/index';
+
+import { wishlistPage } from '../../../fixtures/myAccountMenu/index';
+
+import { assertWishlistHeading, assertEmptyWishlist, assertProductInWishlist, removeProductFromWishlist } from '../../../actions/wishlist/index';
+
+import { categorySweaters, productCarinaCardigan } from '../../../fixtures/categoryPage/index';
+
+import { homePage } from '../../../fixtures/homePage/index';
+
+import { addProductToWishlistFromCategoryPage } from '../../../actions/categoryPage/index';
+
+import { wishistRoute } from '../../../fixtures/wishlist/index';
+
+import { } from '../../../actions/wishlist/index';
+
+import { productValeriaTwoLayeredTankUrl } from '../../../fixtures/productPage/index';
+
+import { addProductToWishlistFromProductPage } from '../../../actions/productPage/index';
+
+
+// TODO add tags CE, EE to test to filter and run tests as needed
+describe('verify single wishlist basic features', () => {
+    it('user should be able to add and remove products from wishlist', () => {
+        visitPage(homePage);
+        openLoginDialog();
+        createAccount(firstName, lastName, accountEmail, accountPassword);
+        assertCreateAccount(firstName);
```

Comment:
I understand why you went for the re-usable function but I feel we should assert in the test instead of using it as a common utility. That's how we do it in the jest tests as well. The reason being, when someone is reading the test, the ultimate thing a test is gonna prove is the assertion and they would like to read the assertion in the test. It gets hard if we don't know what the assertion does.

Thats my view. Thoughts guys?

Reply:
We discussed in chat but I don't know if that was ever applied here - I think we said to go with functions for non assertions, such as `navigateToCart` and `addItemToCart` but to use assertions for things like `didAddItemToCart`

Reply:
We have decided to have assertions also part of functions which makes it easy to maintain in long run, examples can be found here https://github.com/magento/pwa-studio/tree/dev/wishlist-cypress/venia-integration-tests/src/assertions

Reply:
We had a couple of discussions about this in slack and in the Dev time. The decision was to try and omit having assertions as functions but if the assertions are being used too many times and are too many in number, its ok to co-locate them into a function if the developer feels so. 

The important thing is to not overdo it because with multiple developers contributing its hard to keep track of these functions and quickly we can end up with too many assertions and some duplicates.